core/cowdata.h

```diff
@@ -52,29 +52,31 @@ class CowData {
 
 private:
 	mutable T *_ptr;
+	T *_defval;
```

Comment:
I dont think this is a good idea, we should refrain to keep adding stuff here. Also this should be for string, not for cowdata